feat(curriculum): add string method workshop (#66936)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-employee-profile-validator/6a71a8df10e823e77358d896.md

---
id: 6a71a8df10e823e77358d896
title: Step 6
challengeType: 20
dashedName: step-6
---

# --description--

Create another variable named `has_com` that checks if `.com` is in `email`.

> Then print `has_com` so that the output reads: `Has .com: True`.

Reviewer: Same two points as step 5: the variable holds the result of the check rather than performing it, and the trailing punctuation.

```suggestion
Create another variable named `has_com` and assign it the result of checking whether `.com` is in `email`.

Then print `has_com` so that the output reads `Has .com: True`.
```

---
id: 6a71a8df10e823e77358d896
title: Step 6
challengeType: 20
dashedName: step-6
---

# --description--

> Create another variable named `has_com` that checks if `'.com'` is in `email`.

Reviewer 2: ```suggestion
Create another variable named `has_com` that checks if `.com` is in `email`.
```

> Then print both `has_at` and `has_com` with the prefixes `'Has @ symbol: '` and `'Has .com: '` respectively.

Reviewer 2: ```suggestion
Then print the result of concatenating the string`Has .com: ` and the `has_com` variable.
```

> Don't forget to add `str()` to both variables to properly concatenate them with their prefixes.

Reviewer 2: ```suggestion
Don't forget to use the `str()` function to properly convert both numerical variables.
```

# --hints--

You should create a variable named `has_com`.

```js
({ test: () => assert(runPython(`_Node(_code).has_variable('has_com')`)) })
```

Your `has_com` variable should have the value `'.com' in email`.

```js
({ test: () => assert(runPython(`
node = _Node(_code).find_variable('has_com')
not node.is_empty() and (
  node.is_equivalent("has_com = '.com' in email")
)
`)) })
```

You should print `has_at` with the prefix `'Has @ symbol: '`.

```js
({ test: () => assert(runPython(`_Node(_code).has_call("print('Has @ symbol: ' + str(has_at))")`)) })
```

> 

Reviewer 2: Moved to the previous step
```suggestion
```

> You should print `has_com` with the prefix `'Has .com: '`.

Reviewer 2: ```suggestion
You should print the result of concatenating the string `Has .com: ` and the `has_com` variable.
```

```js
({ test: () => assert(runPython(`_Node(_code).has_call("print('Has .com: ' + str(has_com))")`)) })
```

# --seed--

## --seed-contents--

```py
email = 'alice.johnson@company.com'

at_position = email.find('@')
print('Email: ' + email)
print('Position of @: ' + str(at_position))

username = email[:at_position]
print('Username: ' + username)

> has_at = '@' in email

Reviewer 2: ```suggestion
has_at = '@' in email
print('Has @ symbol: ' + str(has_at))
```
